test-kit/components/date-picker-driver.ts

```diff
@@ -0,0 +1,84 @@
+import {DriverBase, selectDom, simulate, trigger} from 'test-drive-react';
+import {DatePicker} from '../../src';
+import {getDayNames} from '../../src/utils';
+
+const bodySelect = selectDom(document.body);
+const datePickerDropdown = 'DATE_PICKER_DROPDOWN';
+
+export class DatePickerTestDriver extends DriverBase {
+    public static ComponentClass = DatePicker;
+
+    public get input(): HTMLInputElement {
+        return this.select<HTMLInputElement>('DATE_PICKER_INPUT');
+    }
+
+    public get selectedDate(): string {
+        return this.select<HTMLInputElement>('DATE_PICKER_INPUT').value;
+    }
+
+    public changeDate(value: string): void {
+        trigger.change(this.input, value);
+        simulate.blur(this.input);
+    }
+
+    public click(elem?: Element): void {
+        if (!elem) {
+            simulate.mouseDown(this.input);
+        } else {
+            simulate.mouseDown(elem);
+        }
+    }
+
+    public openCalender(): void {
+        simulate.focus(this.input);
```

Comment:
This is actually my fault, as the default behavior for opening shouldn't be that it happens on focus. A safer bet is clicking on the calendar icon, which will always cause the calendar to open.

Reply:
done